java-library/src/main/java/uk/co/lucasweb/aws/v4/signer/Signer.java

```diff
@@ -80,32 +78,16 @@ private static String buildAuthHeader(String accessKey, String credentialScope,
     }
 
     private static byte[] hmacSha256(byte[] key, String value) {
-        try {
-            String algorithm = HMAC_SHA256;
-            Mac mac = Mac.getInstance(algorithm);
-            SecretKeySpec signingKey = new SecretKeySpec(key, algorithm);
-            mac.init(signingKey);
-            return mac.doFinal(value.getBytes(UTF_8));
-        } catch (Exception e) {
-            throw new SigningException("Error signing request", e);
-        }
+        return Hmac256Encoder.INSTANCE.encode(key, value);
```

Comment:
This delegates to the `Hmac256Encoder` located in the multiplatform library.

Reply:
I can see that at the moment `Hmac256Encoder` is an `expected object` with just one `actual` implementation for `jvm`. The question is, when we have the implementation for `ios` and `jvm`, will `INSTANCE` resolve to the implementation for the platform that is being compiled?

Reply:
tldr; yes :)

The class we're looking at here is located in the `java-library` module, which [has a dependency](https://github.com/novoda/aws-v4-signer-java/commit/639b390201ade12d5e92e5d8dd830921cfc6f5a0) to the `kmp-library` module. The kotlin multiplatform gradle plugin makes sure that this is resolved to a dependency to the `jvmMain` source set of that library, which contains the `actual` jvm implementation of the `Hmac256Encoder`.

Once the java `Signer` class has been moved to the multiplatform library and migrated to kotlin, the `INSTANCE` prefix will be also gone.